mongo/gridfs/download_stream.go

```diff
@@ -118,8 +118,11 @@ func (ds *DownloadStream) Close() error {
 		return ErrStreamClosed
 	}
 
-	ds.closed = true
-	return nil
+	err := ds.cursor.Close(context.Background())
+	if err != nil {
+		ds.closed = true
```

Comment:
Why are we only setting `closed` to true if the error was non-nil?

Reply:
I assumed if there was an error during Closing, we would not be sure the cursor was actually `closed`.

Reply:
Closing the cursor is a best-effort attempt. At the very least, `ds.cursor.Close()` will release the implicit session that's being held by the cursor. If there's an error running the actual `killCursors` command, we should still set `ds.closed` to true because we tried and there's not really anything else we can do at that point. The cursor will be reaped on the server once it times out.

Reply:
Yep, that sounds good. Thanks for the explanation; I had forgotten about reaping on the server! I do have to check if `ds.cursor` is nil before I close it, as there is a spec test where download length is zero.